### DISCUSSION TITLE
How about that fancy card badge?

### DISCUSSION THREAD

**Discussion by GonzalesMatthew** [nss-evening-cohort-14/evening-client-side#149] [Discuss a Topic]
Regarding our Almost Amazon project, I wanted to do something similar to what we did to our books on sale with regards to displaying our favorite authors, namely: showing a fancy badge icon when our object property 'favorite' === true for an artist. I've narrowed it down to this piece of code and got it work error-free with the below code included in my authors cards:

```
<p class="card-text bold">${item.favorite ? `<span class="badge badge-info sale-badge"><i class="fa heart" aria-hidden="true"></i> Favorite</span>$${item.favorite}` : `$${item.favorite}`} </p>
```

What I don't understand if what exactly is going on here!?
For one (1), our placeholder ${} doesn't close before casting a question mark (?) and additional HTML inside of it. We can do that?! Does anyone know what to Google for this topic? I'd like to learn more, but don't know how to Google this one.
Next up, I tried deleted everything after "span" is closed because I didn't want to see the favorite boolean value printed on my screen because it's ugly and I just want the fancy, cool badge. However, it appears you need this piece of code to tie your hidden badge to the existence of some object property of your choosing? Not including it also triggers a compiling error, so this working solution leaves you with a fancy badge and an ugly boolean value printed on your screen.

So... anyone know more about this? Would love to know what this question mark does, and more on how to tie fancy hidden badges to your card based on object property values the right way.

**Comment by drteresavasquez** [comment]
Do a google search for ternary operator. That's what is being used there. It's like a shorthand if conditional statement. I was wondering if someone would find that and ask about it ☺️

**Reply by GonzalesMatthew** [reply]
Honestly given the struggle bus I was riding today I was surprised I found it, lol.

**Reply by drteresavasquez** [reply]
We are all on some sort of struggle bus. 😊 But this is a good thing to look at if you have bandwidth. We talked about it a little when we did conditionals early on but we will definitely hit it again in the future.

**Comment by drteresavasquez** [comment]
This has a bunch of JS stuff you need to know for React, but look at Item 3. Does that look familiar? https://www-freecodecamp-org.cdn.ampproject.org/c/s/www.freecodecamp.org/news/javascript-skills-you-need-for-react-practical-examples/amp/

**Reply by GonzalesMatthew** [reply]
Thank you! I will give it a read! Thanks for the help
